src/shared/components/palette/dynamic-multi-select-list/read-dynamic-multi-select-list-field.component.spec.ts

```diff
@@ -0,0 +1,166 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { DebugElement } from '@angular/core';
+import { FieldType } from '../../../domain/definition/field-type.model';
+import { CaseField } from '../../../domain/definition/case-field.model';
+import { By } from '@angular/platform-browser';
+import { MockComponent } from 'ng2-mock-component';
+import { ReadDynamicMultiSelectListFieldComponent } from './read-dynamic-multi-select-list-field.component';
+import { text } from '../../../test/helpers';
+import { FixedListPipe } from '../fixed-list/fixed-list.pipe';
+import { FormGroup } from '@angular/forms';
+
+describe('ReadDynamicMultiSelectListFieldComponent', () => {
+
+  const $VALUES = By.css('table>tbody>tr>td');
+
+  const FIELD_ID = 'ReadOnlyFieldId';
+  const FIELD_TYPE: FieldType = {
+    id: 'MultiSelectList',
+    type: 'MultiSelectList',
+    fixed_list_items: [
+      {
+        label: 'Pierre',
+        code: 'P',
+        order: 1
+      },
+      {
+        label: 'Paul',
+        code: 'PA',
+        order: 2
+      },
+      {
+        label: 'Jacques',
+        code: 'J',
+        order: 3
+      }
+    ]
+  };
+  const VALUES = ['P', 'PA', 'J'];
+
+  describe('Non-persistable readonly multi-select-list field', () => {
+    const CASE_FIELD = new CaseField();
+    CASE_FIELD.id = FIELD_ID;
+    CASE_FIELD.label = 'X';
+    CASE_FIELD.field_type = FIELD_TYPE;
+    CASE_FIELD.value = VALUES;
+
+    let FieldReadComponent = MockComponent({
+      selector: 'ccd-field-read',
+      inputs: ['caseField']
+    });
+
+    let fixture: ComponentFixture<ReadDynamicMultiSelectListFieldComponent>;
+    let component: ReadDynamicMultiSelectListFieldComponent;
+    let de: DebugElement;
+
+    beforeEach(async(() => {
+      TestBed
+        .configureTestingModule({
+          imports: [],
+          declarations: [
+            ReadDynamicMultiSelectListFieldComponent,
+            FixedListPipe,
+
+            // Mock
+            FieldReadComponent
+          ],
+          providers: []
+        })
+        .compileComponents();
+
+      fixture = TestBed.createComponent(ReadDynamicMultiSelectListFieldComponent);
+      component = fixture.componentInstance;
+
+      component.caseField = CASE_FIELD;
+
+      de = fixture.debugElement;
+      fixture.detectChanges();
+    }));
+
+    it('render values as a table with one row and one cell per value', () => {
```

Comment:
Additionally, can we have some tests for the _other_ `value` structure - the one where it's an object that contains an array of `list_items` and an array of `value` objects?

Reply:
This might actually be only for the write fields... not sure.